oeps/architectural-decisions/oep-0065/decisions/0002-frontend-app-migrations.rst

```diff
@@ -0,0 +1,152 @@
+Frontend App Migrations
+#######################
+
+Status
+******
+
+Accepted
+
+Summary
+*******
+
+This ADR describes a migration path and simplified repository organization for the ``frontend-app-*`` repositories that is compatible with :term:`Module Architecture`: the :term:`Shell` provided by the `frontend-base <https://github.com/openedx/frontend-base>`_ library, :term:`module federation`, and frontend :term:`projects <Project>`.
+
+Please see :doc:`0001-unified-platform-repository` for more information on ``frontend-base``.
+
+Context
+*******
+
+OEP-65 proposes adopting webpack :term:`module federation` for Open edX :term:`micro-frontends <Micro-frontend>` (MFEs) as a means to enable :term:`runtime module loading` and :term:`shared dependencies` in the :term:`Open edX frontend`. The OEP lays out a series of changes necessary to enable these capabilities, which it refers to as building a "reference implementation" of runtime module loading and shared dependencies.
+
+This reference implementation is effectively a new underlying architecture for our frontend. This ADR refers to this new architecture as the :term:`module architecture`, as opposed to the historical :term:`micro-frontend architecture` that has existed prior to OEP-65.
+
+As part of this paradigm shift, our ``frontend-app-*`` repositories (MFEs) will need to be migrated to work with `frontend-base <https://github.com/openedx/frontend-base>`_. Of particular note, this will require the repositories to:
+
+* Adopt a new set of build/development CLI helpers
+* Use the :term:`shell` to provide the header, footer, and runtime initialization code, amongst other things.
+* Organize their code into loosely-coupled top-level components, which are called :term:`application modules <Application Module>`.
+
+As we adopt ``frontend-base``, the libraries it replaces will undergo their own deprecation processes, which will need to coordinate with the migration of micro-frontends included in Open edX releases. After that deprecation, the micro-frontend architecture will cease to be supported.
+
+Decision
+********
+
+Each of our ``frontend-app-*`` repositories will migrate from being an independent "micro-frontend application" to being a library of modules that can be loaded into a common :term:`Shell`, deployed as a :term:`Site`. These are called :term:`application module libraries <Application Module Library>`. We will document the migration process in detail. At a high level, this will involve the following changes.
```

Comment:
I think I understand why "Application Module Library" (each former MFE will in theory be able to export multiple modules, making it a library), but I have to ask: would trimming it to "Application Module" also work?  It would be easier to remember, pronounce, etc.

Also, if we make the new name too difficult (however accurate), people will just fall back to calling them MFEs.  (Which people probably will anyway.  :shrug:)

Reply:
I think I'm going to drop the word "Application" from it so that it's just 'module library'.  I think we should consider, at some point, whether these repos get renamed to be prefixed with `frontend-library-` or something; I'm not prepared to bite off a real suggestion there yet. :)